boot/seal.go

```diff
@@ -305,12 +360,16 @@ func sealFallbackObjectKeys(key, saveKey keys.EncryptionKey, pbc predictableBoot
 	sealKeyParams := &secboot.SealKeysParams{
 		ModelParams:            modelParams,
 		TPMPolicyAuthKey:       authKey,
-		PCRPolicyCounterHandle: secboot.FallbackObjectPCRPolicyCounterHandle,
+		PCRPolicyCounterHandle: pcrHandle,
 	}
+	logger.Debugf("sealing fallback key with PCR handle: %#x", sealKeyParams.PCRPolicyCounterHandle)
 	// The fallback object contains the ubuntu-data and ubuntu-save keys. The
 	// key files are stored on ubuntu-seed, separate from ubuntu-data so they
 	// can be used if ubuntu-data and ubuntu-boot are corrupted or unavailable.
-	if err := secbootSealKeys(fallbackKeySealRequests(key, saveKey), sealKeyParams); err != nil {
+
+	// XXX find a better name
+	factoryReset := partialReprovisionTPM
```

Comment:
It's not clear which of the variables needs a better name. If it's factoryReset, maybe `doingFactoryReset` or `inFactoryReset`, but then `factoryReset` is equally good IMHO.